libs/sdk-core/src/swap_out/reverseswap.rs

```diff
@@ -541,13 +552,32 @@ impl BTCSendSwap {
         Ok(())
     }
 
-    /// Updates the state of given reverse swap in the cache table, if the status has changed
+    /// Updates the cached values of given reverse swap in the cache table, if the status has changed
     async fn refresh_reverse_swap(&self, rsi: FullReverseSwapInfo) -> Result<()> {
         match self.get_status_update_for_monitored(&rsi).await? {
             None => Ok(()),
-            Some(new_status) => Ok(self
-                .persister
-                .update_reverse_swap_status(&rsi.id, &new_status)?),
+            Some(new_status) => {
+                let lockup_txid = self
```

Comment:
The claim tx is the second transaction (after the lockup) and is constructed and broadcasted in `execute_pending_reverse_swaps` function (which is called right after `refresh_reverse_swap` function call.
With this addition we call the `get_lockup_tx` twice in a row. Why not cache it right after we broadcast it and omit it from here?

Reply:
Makes sense. Applied in [100efb1](https://github.com/breez/breez-sdk/pull/728/commits/100efb106c287e41c5bbf5248ae84ea4ada85064).

Reply:
@danielgranhao  it actually took me a while to understand where exactly we are doing some network requests to get the claim tx or lockup tx regardless of you new code.
I think the flow wasn't clear enough from the code before your code change and your change exposes that.
The problem IMO is that both `refresh_monitored_reverse_swaps` and `execute_pending_reverse_swaps` intersects in their logic for example in `refresh_monitored_reverse_swaps` we are querying the blockchain for the claim tx and doing the same thing also in `execute_pending_reverse_swaps` which follows right after.
So if we want to add a cach layer on to of it we need to change the code first to have a single point where these transactions are fetched from the blockhain and then persisted.

Given the above I think there is no reason for this two steps process (refresh, execute) and we can use one function (currently most of the logic is inside `get_status_update_for_monitored` ) where we:
1. Always fetch both transactions (lock, claim) and cache them
2. Refresh the status and persist it
3. Broadcast the claim transaction if needed (as we are doing in `execute_pending_reverse_swaps` )

It is a bit of refactoring but I think it is necessary. Let me know what do you think.